### PR TITLE
[Term Entry] Dart List .insertAll()

### LINKED ISSUE #4310: [Term Entry] Dart List .insertAll()
### Reason/inspiration (optional)

We would like a new term entry under Dart [List](https://www.codecademy.com/resources/docs/dart/concept/list) for the Dart List `.insertAll()` method. 
Provide the entry in a markdown file under `docs/content/dart/concept/list/terms/insertAll/insertAll.md`

The entry should
    - Have an introductory paragraph about what the method is and does.
    - A syntax section with a pseudo code block with an explanation of it.
    - An example showing the method in use.

Please refer to the [concept entry templates](https://github.com/Codecademy/docs/blob/main/documentation/concept-entry-template.md), [content standards](https://github.com/Codecademy/docs/blob/main/documentation/content-standards.md), and [markdown style guide](https://curriculum-documentation.codecademy.com/content-guidelines/markdown-style-guide/) when working on the PR for this issue.

### Entry Type (select all that apply)

New Term for Existing Concept Entry

### Code of Conduct

- [X] By submitting this issue, I agree to follow Codecademy Doc's [Code of Conduct](https://github.com/Codecademy/docs/blob/main/.github/CODE_OF_CONDUCT.md).

### For Maintainers

- [X] Labels added
- [ ] Issue is assigned

### PR DESCRIPTION
**Pull request by laislemos801**
### Description

- Added the insertAll() term for Dart

### Issue Solved

- Closes #4310 

### Type of Change

- Adding a new entry

### Checklist

- [x] All writings are my own.
- [x] My entry follows the Codecademy Docs style guide.
- [x] My changes generate no new warnings.
- [x] I have performed a self-review of my own writing and code.
- [x] I have checked my entry and corrected any misspellings.
- [x] I have made corresponding changes to the documentation if needed.
- [x] I have confirmed my changes are not being pushed from my forked `main` branch.
- [x] I have confirmed that I'm pushing from a new branch named after the changes I'm making.
- [x] I have linked any issues that are relevant to this PR in the `Issues Solved` section.

### CODE REVIEW

**Review by helfese — APPROVED** [review]

**Review by Sriparno08 — COMMENTED** [review]
Hey @laislemos801, thanks for contributing to Codecademy!

I've listed all the modifications that you need to make in your entry below:

**Review by Sriparno08 — APPROVED** [review]
Looks good for a second review 👍

File: content/dart/concepts/list/terms/insertAll/insertAll.md [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+  - 'Arrays'
+  - 'Collections'
+  - 'Dart'
+  - 'Lists'
+CatalogContent:
+  - 'learn-dart'
+  - 'paths/computer-science'
+---
+
+The **`.insertAll()`** method in Dart is a built-in **List** operation that allows for the insertion of multiple elements into a list at a specified index. While inserting the elements, this method shifts the existing elements as necessary to accommodate the new ones in the list. Also, the specified index should range from _0-N_, where _N_ represents the number of elements that are present in the list.
```
Comment (avdhoottt) [inline content/dart/concepts/list/terms/insertAll/insertAll.md:17]:
```suggestion
The **`.insertAll()`** method in Dart is a built-in **List** method that allows for the insertion of multiple elements into a list at a specified index. While inserting the elements, this method shifts the existing elements as necessary to accommodate the new ones in the list. Also, the specified index should range from _0-N_, where _N_ represents the number of elements that are present in the list.
```

**Review by avdhoottt — APPROVED** [review]
LGTM!!

**Pull request merged**